### PR TITLE
Fix for vxlan_decap test failure on dualtor-aa

### PR DESCRIPTION
**Pull request by AnantKishorSharma**
### Description of PR
vxaln decaptTest adaptation for DToR A/A topology

Summary:
Fixes # https://github.com/sonic-net/sonic-mgmt/issues/11683

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012
- [x] 202205
- [x] 202305
- [ ] 202311

#### What is the motivation for this PR?
vxlan/test_vxlan_decap.py failure on DToR AA with below error
```
FAIL\n\n==================================================================      ====\nFAIL: vxlan-decap.Vxlan\n----------------------------------------------------------------------\nTraceback (most recen      t call last):\n  File \"ptftests/py3/vxlan-decap.py\", line 424, in runTest\n    self.warmup()\n  File \"ptftests/py3/vxlan-      decap.py\", line 358, in warmup\n    raise AssertionError(\"Warmup failed\")\nAssertionError: Warmup failed\n\n-------------      ---------------------------------------------------------\nRan 1 test in 87.954s\n\nFAILED (failures=1)", "rc": 1, "invocati      on": {"module_args": {"creates": null, "executable": null, "_uses_shell": true, "strip_empty_ends": true, "_raw_params": "/r      oot/env-python3/bin/ptf --test-dir ptftests/py3 vxlan-decap.Vxlan --platform-dir ptftests --qlen=10000 --platform remote -t       'vxlan_enabled=False;config_file='\"'\"'/tmp/vxlan_decap.json'\"'\"';count=1;sonic_admin_user='\"'\"'cisco'\"'\"';sonic_admi      n_password='\"'\"'cisco123'\"'\"';sonic_admin_alt_password='\"'\"'YourPaSsWoRd'\"'\"';dut_hostname='\"'\"'1.75.47.10'\"'\"''       --relax --debug info --log-file /tmp/vxlan-decap.Vxlan.NoVxLAN.2024-01-18-16:11:47.log", "removes": null, "argv": null, "wa      rn": true, "chdir": "/root", "stdin_add_newline": true, "stdin": null}}, "start": "2024-01-18 16:11:47.820253", "msg": "non-      zero return code", "stdout_lines": ["Using packet manipulation module: ptf.packet_scapy", "", "*****************************      *************", "ATTENTION: SOME TESTS DID NOT PASS!!!", "", "The following tests failed:", "Vxlan", "", "******************      ************************"], "stderr_lines": ["/root/env-python3/bin/ptf:19: DeprecationWarning: the imp module is deprecated       in favour of importlib; see the module's documentation for alternative uses", "  import imp", "vxlan-decap.Vxlan ... FAIL",       "", "======================================================================", "FAIL: vxlan-decap.Vxlan", "-----------------      -----------------------------------------------------", "Traceback (most recent call last):", "  File \"ptftests/py3/vxlan-d      ecap.py\", line 424, in runTest", "    self.warmup()", "  File \"ptftests/py3/vxlan-decap.py\", line 358, in warmup", "    r      aise AssertionError(\"Warmup failed\")", "AssertionError: Warmup failed", "", "---------------------------------------------
```

#### How did you do it?
Adapted the test for DToR AA.
Part - 1:
vxlan decap test to skip the SoC IPs while generating VLAN Prefixes.

Part - 2:
After Part-1, warm up and "lag to vlan" started passing but "vlan to lag" and vxlan were still failing because expected test pkts were going to the unselected active ToR.
vxlan decap test in A/A topology to consider the possibility of northbound test pkts going to unselected active ToR and check for test pkts on unselected active ToR too.

#### How did you verify/test it?
verified that vxlan/test_vxlan_decap.py passes on T0 DToR AA, AS and standalone setups

#### Any platform specific information?
NA

#### Supported testbed topology if it's a new test case?
NA

### Documentation
NA

### CODE REVIEW

**Comment by AnantKishorSharma** [comment]
@kevinwangsk , @prsunny , @siqbal1986 , could you please review this?

**Comment by siqbal1986** [comment]
can you please add some description as to what the problem was and what has been done to fix it.

**Comment by AnantKishorSharma** [comment]
> can you please add some description as to what the problem was and what has been done to fix it.
Added, please check.

**Review by bpar9 — DISMISSED** [review]

**Review by lolyu — COMMENTED** [review]
why do we need to use the soc IPs?

**Comment by AnantKishorSharma** [comment]
> why do we need to use the soc IPs?

Without this PR we are using. With this PR we are skipping them.

**Comment by jamesan47** [comment]
HI @lolyu , please review and approve if everything looks okay.  And, help with merge to master and double commit to 202305 and 202311.   Thanks!

File: ansible/roles/test/files/ptftests/py3/vxlan-decap.py [hunk 1]
```diff
@@ -483,17 +494,27 @@ def checkRegularRegularVLANtoLAG(self, acc_port, pc_ports, dst_ip, test):
 
         exp_packet = Mask(exp_packet)
         exp_packet.set_do_not_care_scapy(scapy.Ether, "dst")
+        if self.is_active_active_dualtor:
+            exp_packet.set_do_not_care_scapy(scapy.Ether, "src")
```
Comment (kperumalbfn) [inline ansible/roles/test/files/ptftests/py3/vxlan-decap.py:498]: Could you add a comment for masking smac due to aa dualtor.

File: ansible/roles/test/files/ptftests/py3/vxlan-decap.py [hunk 2]
```diff
@@ -169,14 +169,18 @@ def generate_VlanPrefixes(self, gw, prefixlen, acc_ports):
                 else:
                     addr += 1  # skip gw
             res[port] = host_ip
-            addr += 1
+            if self.is_active_active_dualtor:
```
Comment (kperumalbfn) [inline ansible/roles/test/files/ptftests/py3/vxlan-decap.py:172]: Add a comment for skipping the switch ips for aa dualtor.

**Review by kperumalbfn — APPROVED** [review]

**Review by lolyu — APPROVED** [review]

**Comment by lolyu** [comment]
@StormLiangMS, please help review and merge, thanks!

**Review by StormLiangMS — APPROVED** [review]
LGTM

**Pull request merged**